src/index.ts

```diff
@@ -1,105 +1,164 @@
-import { Observable } from 'rxjs';
 import { webContents } from 'electron';
 import * as memoize from 'memoizee';
+// @ts-ignore: no declaration file
+import * as pidtree from 'pidtree';
+// @ts-ignore: no declaration file
+import * as pidusage from 'pidusage';
+import { Observable } from 'rxjs';
 
 import extractURLDomain from './extractURLDomain';
-import { ExtendedProcessMetric } from './index';
 
 // looks like Ectron does not give `getType` to webContents
 interface overridenWebContents extends webContents {
-  getType(): string
+  getType(): string;
 }
 
-export interface onProcessMetricsOptions{ 
+export interface onProcessMetricsOptions {
   /** in ms */
-  samplingInterval?: number
+  samplingInterval?: number;
 }
 
-let getSharedProcessMetricsPoller = (app: Electron.App, samplingInterval: number) => Observable
-  .timer(0, samplingInterval)
-  .map(() => app.getAppMetrics())
-  .share();
+export interface PidUsage {
+  cpu: number;
+  memory: number;
+  ppid: number;
+  pid: number;
+  ctime: number;
+  elapsed: number;
+  timestamp: number;
+}
+
+let getPpid = (pid: number): Promise<number> => {
+  return pidtree(pid, { root: true, advanced: true }).then(([{ ppid }]: PidUsage[]) => ppid);
+};
+
+getPpid = memoize(getPpid, { promise: true });
+
+const getAppUsage = (pid: number): Promise<PidUsage[]> => {
+  return getPpid(pid)
+    .then((ppid: number) => pidtree(ppid, { root: true }))
+    .then(pidusage)
+    .then((usages: any) => Object.values(usages) as PidUsage[]);
+};
+
+let getSharedProcessMetricsPoller = (pid: number, samplingInterval: number) =>
+  Observable.timer(0, samplingInterval)
+    .map(() => Observable.fromPromise(getAppUsage(pid)))
+    .mergeAll()
+    .share();
 
 getSharedProcessMetricsPoller = memoize(getSharedProcessMetricsPoller);
 
 /**
  * Returns an Observable that emits Electron.ProcessMetrics[] on a regular interval.
- * 
+ *
  * For a given `app` and a given `samplingInterval`, the returned observable is shared
  * for performance reasons.
- * 
- * options.samplingInterval = 1000 (1s) by default 
- * @param app 
- * @param options 
+ *
+ * options.samplingInterval = 1000 (1s) by default
+ * @param app
+ * @param options
  */
-export const onProcessMetrics = (app: Electron.App, options: onProcessMetricsOptions) => {
+export const onProcessMetrics = (pid: number, options: onProcessMetricsOptions): Observable<PidUsage[]> => {
```

Comment:
I'd prefer you don't include breaking changes to the external API. Either add retro-compatible changes to this API or add an new API.